packages/config-loader/src/schema/compile.ts

```diff
@@ -27,6 +27,56 @@ import {
 } from './types';
 import { SchemaObject } from 'json-schema-traverse';
 import { normalizeAjvPath } from './utils';
+import { get as getByJsonPointer } from 'jsonpointer';
+
+// Pulled from `json-schema-traverse`. Used to remove non-schema values from JSON pointer.
+const jsonSchemaAncestorKeywords = {
+  additionalItems: true,
+  items: true,
+  contains: true,
+  additionalProperties: true,
+  propertyNames: true,
+  not: true,
+  if: true,
+  then: true,
+  else: true,
+  allOf: true,
+  anyOf: true,
+  oneOf: true,
+  $defs: true,
+  definitions: true,
+  properties: true,
+  patternProperties: true,
+  dependencies: true,
+};
+
+/**
+ * Iterate through the rootSchema in reverse and find all property values in ancestor schemas.
+ * @param rootSchema Primary object.
+ * @param path JSON Pointer path.
+ * @param property Property to collect for.
+ * @returns Set of unique values based on the property.
+ */
+function collectUniqueValuesForPath(
+  rootSchema: object,
+  path: string,
+  property: string,
+) {
+  const parts = path.split('/');
+  const visibilities: Set<ConfigVisibility> = new Set();
+  while (parts.length) {
+    const schema = getByJsonPointer(rootSchema, parts.join('/'));
+    if (schema[property]) {
+      visibilities.add(schema[property]);
+    }
+    parts.pop();
+    if (parts.length && parts[parts.length - 1] in jsonSchemaAncestorKeywords) {
```

Comment:
This logic is a little iffy, but I don't think we have any JSON schema that goes `properties: {
    items: {
        properties: {
            properties: {
                type: 'string'
            }
        }
    }
}`. There's 100% a better way of doing this, but I wasn't able to think of it.